lib/autocomplete.js

```diff
@@ -0,0 +1,90 @@
+const _ = require("lodash");
+const aws = require("aws-sdk");
+const parsers = require("./parsers");
+const core = require("./core");
+const consts = require("./consts.json");
+
+/**
+ * Maps parameters passed to Autocomplete function to an object with parsed values.
+ *
+ * @param params: [ { value: any, name: string, type: string } ]
+ * Array of values passed to autocomplete function as action parameters.
+ *
+ * @returns {*} - object with params names as keys and parsed values
+ */
+function parseActionParameters(params) {
```

Comment:
I do agree with you here, actually specifying the @param type information was the main reason this comment was created -- I just didn't want to leave only this part hanging and went with full java-style doc comment, however if you think the @param part is sufficient, I'm happy to remove the other parts.

I'd argue if the 'parsing' part of the function name is redundant - from my perspective the fact that the mapped result is actually parsed and not in the "raw" format is pretty important. Also this function is  intended for "actionParams" passed to autocomplete functions. While adding "autocomplete" to the name seems a bit excessive (after all it's already in the module exported as "autocomplete") I think it's still worth to highlight it's not intended for any plugin parameters - it's not compatible with the ones passed to the plugin methods. If you really want to drop the "parse" part, my counter-suggestion to `mapPluginParamsToObject` would be something like `mapActionParamsToObject`.

Reply:
The verb is totally fine, I proposed `map` just to shorten the function name, but `parse` is good as well. My point was that `parse` can imply abstract meaning (this function processes string data) and concrete (this function uses parser), and I prefer to avoid releaving the concrete details in function names, but in this case it's just a detail. Let's keep the `parse`.

I'm still missing some business domain knowledge, so some of my naming suggestions might be inaccurate, but `mapActionParamsToObject` is definitely incorrect, because of this:

`  const [params, settings] = [actionParams, pluginSettings].map(parseActionParameters);
`

Reply:
`mapPluginMethodParamsToObject`